packages/compass-preferences-model/src/preferences.ts

```diff
@@ -0,0 +1,318 @@
+   * The telemetryAnonymousId should be used instead.
+   */
+  currentUserId: {
+    type: 'string',
+    required: false,
+    ui: false,
+    cli: false,
+    global: false,
+  },
+  /**
+   * Stores a unique telemetry anonymous ID (uuid) for the current user.
+   */
+  telemetryAnonymousId: {
+    type: 'string',
+    required: true,
+    default: '',
+    ui: false,
+    cli: false,
+    global: false,
+  },
+  /**
+   * Master switch to disable all network traffic
+   * and make Compass behave like Isolated edition always,
+   * i.e. no network traffic other than the one to the db server
+   * (which includes maps, telemetry, auto-updates).
+   */
+  networkTraffic: {
+    type: 'boolean',
+    required: true,
+    default: true,
+    ui: true,
+    cli: true,
+    global: true,
+  },
+  /**
+   * Switch to enable/disable maps rendering.
+   */
+  enableMaps: {
+    type: 'boolean',
+    required: true,
+    default: false,
+    ui: true,
+    cli: true,
+    global: true,
+  },
+  /**
+   * Switch to enable/disable error reports.
+   */
+  trackErrors: {
+    type: 'boolean',
+    required: true,
+    default: false,
+    ui: true,
+    cli: true,
+    global: true,
+  },
+  /**
+   * Switch to enable/disable Intercom panel (renamed from `intercom`).
+   */
+  enableFeedbackPanel: {
+    type: 'boolean',
+    required: true,
+    default: false,
+    ui: true,
+    cli: true,
+    global: true,
+  },
+  /**
+   * Switch to enable/disable usage statistics collection
+   * (renamed from `googleAnalytics`).
+   */
+  trackUsageStatistics: {
+    type: 'boolean',
+    required: true,
+    default: false,
+    ui: true,
+    cli: true,
+    global: true,
+  },
+  /**
+   * Switch to enable/disable automatic updates.
+   */
+  autoUpdates: {
+    type: 'boolean',
+    required: true,
+    default: false,
+    ui: true,
+    cli: true,
+    global: true,
+  },
+};
+
+class Preferences {
+  private _onPreferencesChangedCallbacks: OnPreferencesChangedCallback[];
+  private _userPreferencesModel: PreferencesAmpersandModel;
+
+  constructor(basepath?: string) {
+    // User preferences are stored to disc via the Ampersand model.
+    const PreferencesModel = Model.extend(storageMixin, {
+      props: preferencesProps,
+      extraProperties: 'ignore',
+      idAttribute: 'id',
+      namespace: 'AppPreferences',
+      storage: {
+        backend: 'disk',
+        basepath,
+      },
+    });
+
+    this._onPreferencesChangedCallbacks = [];
+    this._userPreferencesModel = new PreferencesModel();
+  }
+
+  async fetchPreferences() {
+    const userPreferencesModel = this._userPreferencesModel;
+
+    // Fetch user preferences from the Ampersand model.
+    const fetchUserPreferences = promisifyAmpersandMethod(
+      userPreferencesModel.fetch.bind(userPreferencesModel)
+    );
+
+    try {
+      await fetchUserPreferences();
+    } catch (err) {
+      debug('Failed to load preferences, error while fetching models', {
```

Comment:
```suggestion
      // TODO: Log this error with @compass/logging
      debug('Failed to load preferences, error while fetching models', {
```

feels like we should do that, not being able to load preferences sounds like something that could be fairly hard to debug for users